integration_tests/models/test_union_relations.sql

```diff
@@ -0,0 +1,29 @@
+with Union_Three as (
+    {{ pm_utils.union_relations([ref('input_table_1'), ref('input_table_2'), ref('input_table_3')]) }}
+),
+
+Union_Three_Expected as (
+    select 
+    {{ pm_utils.to_integer('1') }} as "RN", 'A' as "Column_A_Expected", {{pm_utils.to_boolean('true')}} as "Column_B_Expected", null as "Column_C_Expected", null as "Column_D_Expected", null as "Column_E_Expected", null as "Column_F_Expected"
+    union all
+    select {{ pm_utils.to_integer('2') }}, 'B', null, {{ pm_utils.to_integer('5') }}, {{ pm_utils.to_double('3.5') }}, null, null
+    union all
+    select {{ pm_utils.to_integer('3') }}, null, {{pm_utils.to_boolean('false')}}, {{ pm_utils.to_integer('7') }}, null, {{ pm_utils.to_boolean('false') }}, null
```

Comment:
The tests can be simplified since 1, 2, 3 are integer by default. You don't need `pm_utils.to_integer()`.
(The boolean do need type casting, because booleans don't exist in SQL Server.)